### PR TITLE
Add test for chain id bug, improve clarity of `_create_fit` code

### PR DESCRIPTION
**Pull request by riddell-stan**
Adds missing test for the chain id reuse bug (already fixed). Also refactors code to make what is going on in the _create_fit function much clearer.

### CODE REVIEW

File: stan/model.py [hunk 1]
```diff
@@ -119,38 +118,40 @@ def fixed_param(self, *, num_chains=4, **kwargs) -> stan.fit.Fit:
...
 
         """
-        assert "chain" not in payload, "`chain` id is set automatically."
-        assert "data" not in payload, "`data` is set in `build`."
-        assert "random_seed" not in payload, "`random_seed` is set in `build`."
-        assert "function" in payload
+        assert "chain" not in kwargs, "`chain` id is set automatically."
+        assert "data" not in kwargs, "`data` is set in `build`."
+        assert "random_seed" not in kwargs, "`random_seed` is set in `build`."
+        assert "function" not in kwargs
```
Comment (ahartikainen) [inline stan/model.py:141]: `function` keyword is now parsed out from `kwargs` by definition, no need to test it here

File: tests/test_basic_bernoulli.py [hunk 2]
```diff
@@ -96,3 +96,20 @@ def test_bernoulli_random_seed_same():
...
+
+
+def test_bernoulli_different_chains(posterior):
+    fit = posterior.sample(num_chains=4)
+    assert fit.num_chains == 4
+
+    # for a fit with only one scalar parameter, it is the last one
+    theta_chain_1 = fit._draws[-1, :, 0]
+    theta_chain_2 = fit._draws[-1, :, 1]
+    assert not np.allclose(theta_chain_1, theta_chain_2)
```
Comment (ahartikainen) [inline tests/test_basic_bernoulli.py:108]: If we sample 4 chains, maybe we can test all pairs?

**Review by ahartikainen — CHANGES_REQUESTED** [review]

File: stan/model.py (hunk 1 above)
Comment (riddell-stan) [reply stan/model.py]: Fixed.

File: tests/test_basic_bernoulli.py (hunk 2 above)
Comment (riddell-stan) [reply tests/test_basic_bernoulli.py:108]: Fixed.

**Review by ahartikainen — APPROVED** [review]

**Pull request merged**